### DISCUSSION TITLE
How to track dividend?

### DISCUSSION THREAD

**Discussion by skorvek** [ghostfolio/ghostfolio#1001] [Q&A]
How are divdends treated in ghostfolio?

I have both individual stocks/ETFs and mutual funds tracked. How should dividend reinvestment be handled to avoid double-counting?  I don't see any cash-value balance under an account (that isn't configured as a cash/savings account). We aren't adding cash to a brokerage account and then withdrawing it to do a purchase- this isn't accounting software.

Also one of my accounts- a traditional brokerage/margin account with a cash balance - lists a dividend simply as a cash amount.  In Ghostfolio, i need to enter a quantity and price, which leads so some imprecise division to arrive at an entry every quarter. And of course it is cash, not a stock purchase.

In the mutual fund account, the dividends are of course automatically reinvested as another share purchase. So that could be simply entered as another buy of shares of the fund to avoid double-counting. But then it doesn't appear as a dividend.

What's the correct method?

**Comment by skorvek** [comment]
wait- I do see a cash balance column under the accounts tab- so is the answer to record dividends for a brokerage account simply record a buy for a mutual fund?

**Comment by dtslvr** [comment]
Hello @skorvek and welcome to the 👻  Ghostfolio community!

Dividend have no effect on your total wealth, you can track them as an activity of type `DIVIDEND`. In the corresponding account you need to keep the (cash) balance up to date or track another `BUY` activity if you reinvest the amount.

I don't know how mutual funds work, but ETFs are either distributing (track dividend as described above) or accumulating (you don't have to track because it's included in the market price).

Does that help you a little bit?

**Comment by skorvek** [comment]
Think of a mutual fund as an ETF with a DRIP auto-purchase plan. That's the default option for most.

So I se that if I whip out the calculator and add up all the buys/sells of a symbol excluding dividends it will match the total at the bottom of the value column. But then why do you show the dividends in the table if the total row doesn't include all the transaction rows? 

Putting funds aside- if I had a divided into my margin account- I would now have a positive cash balance. That I do not see if I go to the account listing. I'll see the value of the investments but no cash.

**Comment by skorvek** [comment]
In other stock tracking applications, and any accounting software (which I understand this is not) you would link the cash account and the portfolio. So thus when a dividend is paid (or a fee is incurred) a transaction would correspondingly be made in the currency account. 

Your UI makes is appear that those are indeed being tracked in one account as per the account page and the activities page per security, but sounds like it is not.  I would suggest splitting the table on the account listing page by account type so you can show different columns: i.e hide the cash balance column for securities accounts, hide the value column (or reuse that label for the balance) for a cash account.

Also on the performance graph modal for a security change the label from "total" to "security value" or "holding value" or something similar so it doesn't appear as a column total.

**Reply by fib1985** [reply]
Hi, you must switch to portfolio performance app instead. Believe me...

**Comment by nlighteneddesign** [comment]
To put it more simply, I have a mutual fund in a brokerage that I mark as having reinvestment.
Whenever dividends are applied it correlates to a buy action.

If I record those as dividends in Ghostfolio then the totals are quickly wrong because Ghostfolio does not convert those to new shares and the growth is off.
I manually converted the dividends to a buy action instead with a note of dividend to confirm and Ghostfolio reports the correct value at that point. It would be great to be able to mark a fund as a reinvestment fund where dividends are automatically converted into purchases..

Ultimately in the overview what I care about is the total value so I can just record them as buys, but that is technically not correct and I would have no way to get a dividend report.

If you can point to where I can contribute this, I can take a look.

**Comment by Citrus0299** [comment]
After going through all the options I found out that there is currently no way to accuratly track the **individual** performance of a mutual fund with external dividens in ghostfolio.

The only accurate workaround I could find is to create a `BUY` order with the amount of stock you received for the dividend with a close to zero unit price e.g. `0.00001`.
This will result in an accurate `ROI` performance calculation. However Ghostfolio currently uses the `ROAI` performance calculation method which will be increased by such a low unit price. With experimental features enabled there is at least a disabled selector box to 
hopefully select `ROI` in the future.

### Why can't I just use a normal `BUY` order with the unit price?
This will increase your invested amount in the fund and therefore the individual performance will be less.

### Why can't I just use a `BUY` and `DIVIDEND` order?
Same reason as above and you will also log dividends that never really happened as you just received stock and no cash.
This will result in a wrong performance and total dividend calculation.

### How can this be fixed?
Allow a `DIVIDEND` to pay out stock instead of only cash or workaround the problem by allowing a `BUY` with a zero unit price which will be ignored.


I hate using JS on the server from the bottom of my heart so I can only hope someone else will implement a proper solution.

**Reply by ptoulouse** [reply]
One can argue that a dividend and a buy  order is the correct way to account for the transaction. The company issued a dividend, and you just happened to decide to buy new shares with that dividend in advance.

I would also argue that this is the way the government sees it (at least in my jurisdiction) as reinvested dividend is taxable as a regular dividend.

**Reply by Citrus0299** [reply]
My argument was specifically about **mutual funds** and not single stocks, what you are talking about is completely different.
Yes you are right, if a **single stock** issues a dividend you get cash and have to pay taxes for it and you can **decide** to reinvest it.

However with a mutual fund you never receive cash, just more stock of the fund. Because you never received cash you don't have to pay taxes and cant't choose to reinvest. This is fundamentally different as I already explained.

**Reply by Citrus0299** [reply]
There are three ways funds can issue dividends they received from the companies (single stocks) inside the fund:
1. Pay it out in cash to the shareholders (trackable)
2. Pay it out in stock to the shareholders (untrackable in ghostfolio)
3. Buy more shares of the companies in the fund therefore increasing the value of the fund (no tracking required)

**Reply by ptoulouse** [reply]
Even for mutual funds, and even if your not receiving money, the dividend is still taxable in Canada (unless they are in a tax-sheltered account), and apparently so in the US as well: [How Do You Pay Taxes on a Fund That Reinvests Dividends?](https://www.investopedia.com/ask/answers/011215/if-i-reinvest-my-dividends-are-they-still-taxable.asp#toc-are-reinvested-dividends-taxable)
